app/javascript/src/controller_form_analytics.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Form Analytics Controller
+ * ----------------------------------------------------
+ * Description:
+ * Adds functionality for the Settings:Analytics FB Editor view.
+ * File and controller name based on Backend controller name.
+ *
+ * Documentation:
+ *
+ *     - Requires jQuery & jQueryUI
+ *       https://api.jquery.com/
+ *       https://api.jqueryui.com/
+ *
+ *     - TODO:
+ *       (steven.burnell@digital.justice.gov.uk to add).
+ *
+ **/
+
+
+const Expander = require('./component_expander');
+const DefaultController = require('./controller_default');
+
+
+class FormAnalyticsController extends DefaultController {
+  constructor(app) {
+    super(app);
+    switch(app.page.action) {
+      case "create":
+      case "index":
+        this.index();
+        break;
+    }
+  }
+
+
+  /* VIEW ACTION
+   **/
+  index() {
+    this.#addExpanderEnhancement();
+  }
+
+
+  /* VIEW SETUP
+   * 1. Enhance native browser (<details>) functioality with Expander.
+   * 2. Make sure the Expanders will be open if Checkbox ticked or an error shows.
+   * 3. Allow Checkboxes to also control the Expander components.
+   **/
+  #addExpanderEnhancement() {
+    $(".analytics-environment-configuration").each(function(index) {
+      var $this = $(this);
+      var $checkbox = $("input[type=checkbox]", $this);
+      var expander = new Expander($("details", $this), {
+        auto_open: $(".govuk-form-group--error", $this).length
+      });
+
+      $checkbox.on("click", function() {
+        if(this.checked && !expander.isOpen()) {
+          expander.open();
+        }
+      });
```

Comment:
In here you could grab the `id` of the `Expander` component, and add an `aria-controls` attribute to the checkbox input, and also add an `aria-expanded` attribute on checking the box (and remove when unchecking)

```
$checkbox.attr('aria-controls', $("details", $this).attr('id'));
```

Not sure how great it is to have two things controlling the same widget accessibility wise, but it is probably best to let users know that something is going to happen when they check the box.

Reply:
We could but that might be confusing. As mentioned above, the Checkbox is not really a control. Nothing happens when it is unchecked, so removing 'expanded' would not be correct. The actual control, `<summary>`, is presumably doing all that but, I believe, its 'expanded' value does get updated by the Checkbox, so that it is kept in sync.

Actually, I doubt the native functionality of details/summary even needs any aria attribute for this, but it's there because the open method simply sets the state, and the state updates the actual activator, which is not always a `<summary>` tag. I'm taking the position of 'no harm in having both' and 'saves having extra browserControlled checks around that attribute update'.

Reply:
I'll respond to both comments here:

The TLDR is that I think I agree and that on reflection the best solution* is not to add the extra attributes.

You're right in that the checkbox is not a 'full' control in that it only opens, and doesn't close the expander.  

However, I think your assumption of how a screenreader user navigates the page is possibly slightly off.  My understanding is that they won't naively tab through a document like a sighted user, they will often 'scan' a page before navigating it - to find all headings, or form elements or interactive elements - so if they don't know more content has been revealed (by a screenreader announcing an element controlling an expanding area) then they may not know it is there, and continue with their mental model of the page as they've scanned it.

So in this instance a screenreader user may have discovered the forms with the checkboxes and submit button. So after checking the box, may just hit return to submit the form.

In this instance, as it is a weird case, I think we'll err on not adding the extra aria - but there is the potential of non-discovery of content.

On another note, adding aria attributes to a native details summary is almost certainly an anti-pattern, as the best aria is no aria! (where possible of course) and a native details/summary shouldn't need aria.

* In actual fact I think the best solution here is that we should't have both the checkbox reveal and the details summary.  The checkbox state should just control the visibility of the config.  Even if they're both enabled it's only 6 fields, which is not that many, and that would be a simple, easy to understand interface for all users (and consistent with other areas of the UI too)